lib/engine/game/g_18_mag/step/route.rb

```diff
@@ -35,8 +35,16 @@ def process_run_routes(action)
             raise GameError, 'Must use all purchased rail-car benefits' unless @game.all_railcars_used?(action.routes)
 
             ciwl_income(action.routes) if @game.new_major?
-
             super
+            deconvert_plus_one_train if @game.supporters? && @round.original_train
+          end
+
+          def deconvert_plus_one_train
+            @round.ma_plus_train.name = @round.original_train.name
+            @round.ma_plus_train.distance = @round.original_train.distance
+
+            @round.original_train = nil
+            @round.ma_plus_train = nil
```

Comment:
What is this doing? Why are you assigning to @round.ma_plus_train and then making it nil?

Reply:
see here: https://github.com/tobymao/18xx/blob/master/lib/engine/game/g_1822/step/route.rb#L50

I'm modifying the distance of the train that was converted into +1 town range back to it's original values.
That's why i'm change .name and . distance, then setting it to nil.
Similar code to pullman attachment and removal in 1822